test/integration/node-specific/bson-options/utf8_validation.test.ts

```diff
@@ -1,125 +1,52 @@
-    bsonSpy = null;
-  });
-
   let client;
 
   afterEach(async () => {
+    sinon.restore();
     if (client) await client.close();
   });
 
-  describe('enableUtf8Validation option set to false', () => {
-    const option = { enableUtf8Validation: false };
-
-    for (const passOptionTo of ['client', 'db', 'collection', 'operation']) {
-      it(`should disable validation with option passed to ${passOptionTo}`, async function () {
-        client = this.configuration.newClient(passOptionTo === 'client' ? option : undefined);
+  context(
+    'when the server is given a long multibyte utf sequence and there is a writeError that includes invalid utf8',
+    () => {
+      let client: MongoClient;
+      let error: MongoServerError;
+      beforeEach(async function () {
+        client = this.configuration.newClient();
 
-        const db = client.db('bson_utf8Validation_db', passOptionTo === 'db' ? option : undefined);
-        const collection = db.collection(
-          'bson_utf8Validation_coll',
-          passOptionTo === 'collection' ? option : undefined
-        );
+        async function generateWriteErrorWithInvalidUtf8() {
+          // Insert a large string of multibyte UTF-8 characters
+          const _id = '\u{1F92A}'.repeat(100);
 
-        await collection.insertOne(
-          { name: 'John Doe' },
-          passOptionTo === 'operation' ? option : {}
-        );
+          const test = client.db('parsing').collection<{ _id: string }>('parsing');
+          await test.insertOne({ _id });
 
-        expect(bsonSpy).to.have.been.called;
-        const result = bsonSpy.lastCall.returnValue;
-        expect(result).to.deep.equal(EXPECTED_VALIDATION_DISABLED_ARGUMENT);
-      });
-    }
-  });
+          const spy = sinon.spy(OpMsgResponse.prototype, 'parse');
 
-  describe('enableUtf8Validation option set to true', () => {
-    // define client and option for tests to use
-    const option = { enableUtf8Validation: true };
-    for (const passOptionTo of ['client', 'db', 'collection', 'operation']) {
-      it(`should enable validation with option passed to ${passOptionTo}`, async function () {
-        client = this.configuration.newClient(passOptionTo === 'client' ? option : undefined);
-        await client.connect();
-
-        const db = client.db('bson_utf8Validation_db', passOptionTo === 'db' ? option : undefined);
-        const collection = db.collection(
-          'bson_utf8Validation_coll',
-          passOptionTo === 'collection' ? option : undefined
-        );
-
-        await collection.insertOne(
-          { name: 'John Doe' },
-          passOptionTo === 'operation' ? option : {}
-        );
-
-        expect(bsonSpy).to.have.been.called;
-        const result = bsonSpy.lastCall.returnValue;
-        expect(result).to.deep.equal(EXPECTED_VALIDATION_ENABLED_ARGUMENT);
-      });
-    }
-  });
+          error = await test.insertOne({ _id }).catch(error => error);
```

Comment:
It looks like we only test the default behavior here, could we add a case with utf8 validation explicitly enabled also?